CRM/Utils/Recent.php

```diff
@@ -188,4 +204,51 @@ public static function delContact($id) {
     $session->set(self::STORE_NAME, self::$_recent);
   }
 
+  /**
+   * Check if a provider is allowed to add stuff.
+   * If correspondig setting is empty, all are allowed
+   *
+   * @param string $providerName
+   */
+  public static function isProviderEnabled($providerName) {
+
+    // Join contact types to providerName 'Contact'
+    $contactTypes = CRM_Contact_BAO_ContactType::contactTypes(TRUE);
+    if (in_array($providerName, $contactTypes)) {
+      $providerName = 'Contact';
+    }
+    $allowed = TRUE;
+
+    // Use core setting recentItemsProviders if configured
+    $providersPermitted = CRM_Core_BAO_Setting::getItem(CRM_Core_BAO_Setting::SYSTEM_PREFERENCES_NAME, 'recentItemsProviders');
+    if ($providersPermitted) {
+      $allowed = in_array($providerName, $providersPermitted);
+    }
+    // Else allow
+    return $allowed;
+  }
+
+  /**
+   * Gets the list of available providers to civi's recent items stack
+   */
+  public static function getProviders() {
+    $providers = array(
+      'Contact' => ts('Contacts'),
+      'Relationship' => ts('Relationships'),
+      'Activity' => ts('Activities'),
+      'Note' => ts('Notes'),
+      'Group' => ts('Groups'),
+      'Case' => ts('Cases'),
+      'Contribution' => ts('Contributions'),
+      'Participant' => ts('Participants'),
+      'Grant' => ts('Grants'),
+      'Membership' => ts('Memberships'),
+      'Pledge' => ts('Pledges'),
+      'Event' => ts('Events'),
+      'Campaign' => ts('Campaigns'),
+    );
```

Comment:
Yeah, I did this first, but i removed it for different reasons:
1. Syncing: it's somehow different to keep in sync and en-/disabling components will cause problems: If you disabled a component A, head over to this setting, select some enabled components B and C and reenable component A, you'll get no items of this one by default. And if your users don't miss it, they will never know it would be possible.
1. Overkill: I implemented the permission check in Recent.php's add method and used it's type param. Components have different names. It's possible to map them, but regarding 1. I thought it would be too much thrill.

But if you want, I'll do...